### PR TITLE
Operator Development celu and celu_

### PR DESCRIPTION
**Pull request by yuxingcyx**
### PR Category

Operator, OP Test, Benchmark
### Type of Change

New Feature
### Description

Implement celu and celu_ operator
### Issue
https://github.com/FlagOpen/FlagGems/issues/887

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance

both celu and celu_ pass accuracy test:
image

celu performance：
image
image
image
celu_ performance:
image
image
image

### CODE REVIEW

File: tests/test_unary_pointwise_ops.py [hunk 1]
```diff
@@ -453,6 +453,40 @@ def test_accuracy_elu_backward(shape, dtype):
...
+    with flag_gems.use_gems():
+        res_out = torch.nn.functional.celu(inp, alpha)
+
+    gems_assert_close(res_out, ref_out, dtype)
+
+
+@pytest.mark.inplace
+@pytest.mark.celu_
+@pytest.mark.parametrize("shape", POINTWISE_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
```
Comment (meinie0826) [inline tests/test_unary_pointwise_ops.py:475]: `alpha` should also be tested.

File: tests/test_unary_pointwise_ops.py [hunk 2]
```diff
@@ -453,6 +453,40 @@ def test_accuracy_elu_backward(shape, dtype):
...
+def test_accuracy_celu_(shape, dtype):
+    inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
+    alpha = torch.rand(1).item()
+
+    res_inp = inp.clone().to(flag_gems.device)
```
Comment (meinie0826) [inline tests/test_unary_pointwise_ops.py:480]: no need to do that

File: src/flag_gems/ops/celu.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    logger.debug("GEMS CELU")
+    n_elements = A.numel()
+    output = torch.empty_like(A)
+    grid = lambda meta: (triton.cdiv(n_elements, meta['BLOCK_SIZE']), )
+    celu_forward_kernel[grid](
+        A, output, alpha, scale, input_scale, n_elements, BLOCK_SIZE=2048
+    )
+    return output
+
+def celu_(A, alpha=1.0, scale=1.0, input_scale=1.0):
```
Comment (meinie0826) [inline src/flag_gems/ops/celu.py]: It seems that torch does not have the parameters `scale` and `input_stcale`. Could you please confirm again? By the way, can we consider implementing this operator using `pointwise dynamic`? 
like this:https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/ops/elu.py

Comment (yuxingcyx) [reply src/flag_gems/ops/celu.py]: Thanks for your reply. Yes, torch does not have these parameters. These two parameters I wrote with reference to elu.py actually.  I used pointwise dynamic first, but the performace is better after using the code I handed in. Do I still need to use pointwise dynamic to write the code? Thank you.

Comment (meinie0826) [reply src/flag_gems/ops/celu.py]: Function signatures are not based on torch, but are aligned with aten, see `pytorch/aten/src/ATen/native/native_functions.yaml`

Comment (meinie0826) [reply src/flag_gems/ops/celu.py]: `Pointwised_dynamic` aims to solve the problem of generating code for high-dimensional tensors that are discontinuous. Handwritten writing not only makes it difficult to handle continuous tensors, but also introduces additional overhead. Therefore, it is recommended to use continuous tensors

Comment (yuxingcyx) [reply src/flag_gems/ops/celu.py]: For scale and input_scale, you suggest me to delete these 2 parameters. Is that what you mean?

Comment (yuxingcyx) [reply src/flag_gems/ops/celu.py]: @meinie0826 Btw, I saw "unable to access github" in my checks. Could you plz solve this? thx
image

Comment (meinie0826) [reply src/flag_gems/ops/celu.py]: > For scale and input_scale, you suggest me to delete these 2 parameters. Is that what you mean?

Yes, I think these two parameters are not consistent with torch and should be aligned with torch

Comment (meinie0826) [reply src/flag_gems/ops/celu.py]: > @meinie0826 Btw, I saw "unable to access github" in my checks. Could you plz solve this? thx image

dont care about that, modify the code first

Comment (yuxingcyx) [reply src/flag_gems/ops/celu.py]: Using pointwise dynamic makes the performace worse. 
image

Comment (yuxingcyx) [reply src/flag_gems/ops/celu.py]: @meinie0826 I have implemented the decorator pointwise_dynamic to code. Plz review. Thx

**Review by meinie0826 — DISMISSED** [review]
lgtm

**Review by StrongSpoon — DISMISSED** [review]
lg

**Comment by yuxingcyx** [comment]
> please reformat your code. everything else is fine.

ok, I reformated the code.

**Comment by StrongSpoon** [comment]
It appears that format error still exists. I suggest correcting it on your device according to https://github.com/FlagOpen/FlagGems/blob/master/docs/code_countribution.md#code-format-check.

**Review by StrongSpoon — APPROVED** [review]
lg
